src/helpers/checks.jl

```diff
@@ -1,20 +1,94 @@
+"""
+    prepare_check_result(log_range, errors, slope)
+
+Given a range of values `log_range`, where we computed `errors`,
+check whether this yields a slope of `slope` in log-scale
+
+Note that if the errors are below the given tolerance and the method is exact,
+no plot will be generated,
+
+# Keyword arguments
+
+* `exactness_tol` - (`1e3*eps(eltype(errors))`) is all errors are below this tolerance, the check is considered to be exact
+* `io` – (`nothing`) provide an `IO` to print the check result to
+* `name` (`"differntial"`) – name to display in the check (e.g. if checking gradient)
+* `plot`- (`false`) whether to plot the resulting check (if `Plots.jl` is loaded). The plot is in log-log-scale. This is returned and can then also be saved.
+* `slope_tol` – (`0.1`) tolerance for the slope (global) of the approximation
+* `throw_error` - (`false`) throw an error message if the gradient is wrong
+"""
+function prepare_check_result(
+    log_range,
+    errors,
+    slope;
+    io::Union{IO,Nothing}=nothing,
+    name="estimated slope",
+    slope_tol=1e-1,
+    plot=false,
+    throw_error=false,
+    window=nothing,
```

Comment:
Window is not documented yet.

Reply:
Oh, that one I have to check tomorrow then, because that is from the first implementation from Constantin still. I even added a few docs that were missing already.

Reply:
Ah, found it :)